### DISCUSSION TITLE
Does the Assimp library support non-skeletal animation?

### DISCUSSION THREAD

**Discussion by suikameronn** [assimp/assimp#5792] [Q&A] [answered]
Does the Assimp library support non-skeletal animation?

I want to load an FBX file that records non-skeletal animation using Assimp and play it back. Non-skeletal animation refers to animation where the model does not have a skeleton or bones attached.

I would like to load and play a 3D model like the following using assimp.

https://github.com/user-attachments/assets/7626e5ca-67eb-4c2d-8a21-e5dbe63b06a7

**Comment by kimkulling** [comment, marked as answer]
Yes, you can use the node animations to model this:

```
aiNode *node = new aiNode;
node->mName.Set("node");
aiNodeAnim anim;
anim.mNodeName("node");
anim.mNumPositionKeys = 2;
anim.mPositionKeys = new aiVectorKey[anum.mNumPositionKeys];
```
You can describe the animation steps with the vector keys. Each key has a time and a position. If you want to change the interpolation mode: this is also supported.

**Reply by AMZN-Gene** [reply]
Is there a way to view animations using assimp_view?
**Edit:** Ahh, I see... there's a built-in [assimp_view](https://github.com/assimp/assimp/blob/ce0a50ec21bfa4bb2535a4fbd87b67feadc0cea0/tools/assimp_view/CMakeLists.txt#L51). I was using a different [assimp_view](https://github.com/assimp/assimp_view)
